Std/Data/Fin/Lemmas.lean

```diff
@@ -644,7 +644,7 @@ theorem exists_fin_two {p : Fin 2 → Prop} : (∃ i, p i) ↔ p 0 ∨ p 1 :=
   exists_fin_succ.trans <| or_congr_right exists_fin_one
 
 theorem fin_two_eq_of_eq_zero_iff : ∀ {a b : Fin 2}, (a = 0 ↔ b = 0) → a = b := by
-  simp [forall_fin_two]
+  simp [forall_fin_two]; decide
```

Comment:
This probably counts as a nonterminal simp.

Reply:
I would say it's an exception, `decide` is one of those tactics that is not very sensitive to the result of `simp`. Although, it seems like this could just be `by decide` without the `simp`?

Reply:
Unfortunately just `by decide` returns
```
failed to synthesize
  Decidable (∀ {a b : Fin 2}, (a = 0 ↔ b = 0) → a = b)
```

Reply:
Either way, I replaced `simp` by `simp only` just to be sure.